### ISSUE TITLE
Analyzing Monotonic Linear Interpolation in Neural Network Loss Landscapes

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#86]
https://arxiv.org/abs/2104.11044

Tronto, Vectorの研究とても良さそう

初期のニューラルネットワークパラメータと、確率的勾配降下法（SGD）によるトレーニング後の収束パラメータとの間の線形補間（Monotonic Linear Interpolation: MLI）は、典型的には、トレーニング目的の単調な減少をもたらす。
ニューラルネットワークの非凸の目的と高度に非線形な学習ダイナミクスにもかかわらず、この特性は多くに見られる。

この特性が成り立つ条件を導出した（この条件はよく使われるデータセットモデルにおいて成り立つ）

<img width="711" alt="Screen Shot 2021-04-27 at 23 00 30" src="https://user-images.githubusercontent.com/8721858/116255586-57412500-a7ad-11eb-9bb5-978cf44a226c.png">
<img width="1236" alt="Screen Shot 2021-04-27 at 23 00 43" src="https://user-images.githubusercontent.com/8721858/116255600-5ad4ac00-a7ad-11eb-8c17-3f69f37a7135.png">
<img width="612" alt="Screen Shot 2021-04-27 at 23 01 12" src="https://user-images.githubusercontent.com/8721858/116255602-5c05d900-a7ad-11eb-8c79-e8566f1a45c3.png">

> Linear interpolation between initial neural network parameters and converged parameters after training with stochastic gradient descent (SGD) typically leads to a monotonic decrease in the training objective. This Monotonic Linear Interpolation (MLI) property, first observed by Goodfellow et al. (2014) persists in spite of the non-convex objectives and highly non-linear training dynamics of neural networks. Extending this work, we evaluate several hypotheses for this property that, to our knowledge, have not yet been explored. Using tools from differential geometry, we draw connections between the interpolated paths in function space and the monotonicity of the network - providing sufficient conditions for the MLI property under mean squared error. While the MLI property holds under various settings (e.g. network architectures and learning problems), we show in practice that networks violating the MLI property can be produced systematically, by encouraging the weights to move far from initialization. The MLI property raises important questions about the loss landscape geometry of neural networks and highlights the need to further study their global properties.
